python/tvm/driver/tvmc/runner.py

```diff
@@ -0,0 +1,450 @@
+        print(top_results)
+
+    if args.outputs:
+        # Save the outputs
+        np.savez(args.outputs, **outputs)
+
+
+def get_input_info(graph_str, params):
+    """Return the 'shape' and 'dtype' dictionaries for the input
+    tensors of a compiled module.
+
+    Parameters
+    ----------
+    graph_str : str
+        JSON graph of the module serialized as a string.
+    params : bytearray
+        Params serialized as a bytearray.
+
+    Returns
+    -------
+    shape_dict : dict
+        Shape dictionary - {input_name: tuple}.
+    dtype_dict : dict
+        dtype dictionary - {input_name: dtype}.
+    """
+    # NOTE - We can't simply get the input tensors from a TVM graph
+    # because weight tensors are treated equivalently. Therefore, to
+    # find the input tensors we look at the 'arg_nodes' in the graph
+    # (which are either weights or inputs) and check which ones don't
+    # appear in the params (where the weights are stored). These nodes
+    # are therefore inferred to be input tensors.
+
+    shape_dict = {}
+    dtype_dict = {}
+    # Use a special function to load the binary params back into a dict
+    load_arr = tvm.get_global_func("tvm.relay._load_param_dict")(params)
+    param_names = [v.name for v in load_arr]
+    graph = json.loads(graph_str)
+    for node_id in graph["arg_nodes"]:
+        node = graph["nodes"][node_id]
+        # If a node is not in the params, infer it to be an input node
+        name = node["name"]
+        if name not in param_names:
+            shape_dict[name] = graph["attrs"]["shape"][1][node_id]
+            dtype_dict[name] = graph["attrs"]["dltype"][1][node_id]
+
+    logger.debug("collecting graph input shape and type:")
+    logger.debug("graph input shape: %s", shape_dict)
+    logger.debug("graph input type: %s", dtype_dict)
+
+    return shape_dict, dtype_dict
+
+
+def generate_tensor_data(shape, dtype, fill_mode):
+    """Generate data to produce a tensor of given shape and dtype.
+
+    Random data generation depends on the dtype. For int8 types,
+    random integers in the range 0->255 are generated. For all other
+    types, random floats are generated in the range -1->1 and then
+    cast to the appropriate dtype.
+
+    This is used to quickly generate some data to input the models, as
+    a way to check that compiled module is sane for running.
+
+    Parameters
+    ----------
+    shape : tuple
+        The shape of the tensor.
+    dtype : str
+        The dtype of the tensor.
+    fill_mode : str
+        The fill-mode to use, either "zeros", "ones" or "random".
+
+    Returns
+    -------
+    tensor : np.array
+        The generated tensor as a np.array.
+    """
+    if fill_mode == "zeros":
+        tensor = np.zeros(shape=shape, dtype=dtype)
+    elif fill_mode == "ones":
+        tensor = np.ones(shape=shape, dtype=dtype)
+    elif fill_mode == "random":
+        if "int8" in dtype:
+            tensor = np.random.randint(256, size=shape, dtype=dtype)
```

Comment:
let us restrict to 128. If the model is int8 (not uint8). 256 will make us overflow.

Reply:
fixed